src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -0,0 +1,472 @@
+                            }
+                            return Tuple.Create(TheAction.Move, i, idx); // pass the new position and old position
+                        }
+                    }
+                    // the element doesn't exist yet. If there's a comparer and other items, add it in the correct spot
+                    else if (comparer != null && list.Count > 0)
+                    {
+                        var i = 0;
+                        for (; i < list.Count && comparer(t, list[i]) > 0; i++) { }
+                        if (i == list.Count)
+                        {
+                            list.Add(t);
+                            return Tuple.Create(TheAction.Add, list.Count - 1, -1);
+                        }
+                        list.Insert(i, t);
+                        return Tuple.Create(TheAction.Insert, i, -1);
+                    }
+                    else
+                    {
+                        list.Add(t);
+                        return Tuple.Create(TheAction.Add, list.Count - 1, -1);
+                    }
+                })
+                .ObserveOn(scheduler)
+                .Select(x =>
+                {
+                    UpdateFilter(x.Item1, x.Item2, x.Item3);
+                    return Unit.Default;
+                })
+                .Publish()
+                .RefCount();
+            return this;
+        }
+
+        public void SetComparer(Func<T, T, int> compare)
+        {
+            comparer = compare;
+            var list = filter == null ? this : original as IList<T>;
+            UpdateSort(list, 0, list.Count);
+            UpdateFilter(filter);
+        }
+
+        public void SetFilter(Func<T, int, IList<T>, bool> theFilter)
+        {
+            UpdateFilter(theFilter);
+        }
+
+        void UpdateSort(IList<T> list, int start, int end)
+        {
+            if (comparer == null)
+                return;
+
+            for (; start < end - 1; start++)
+            {
+                var idxSmallest = start;
+                for (var i = start + 1; i < end; i++)
+                    if (comparer(list[i], list[idxSmallest]) < 0)
+                        idxSmallest = i;
+                if (idxSmallest == start) continue;
+                Swap(list, start, idxSmallest);
+
+                // if there's a filter, then it's going to trigger events and we don't need to manually trigger them
+                if (filter == null)
+                {
+                    OnPropertyChanged(new PropertyChangedEventArgs("Item[]"));
+                    OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Move, this[start], start, idxSmallest));
+                }
+            }
+        }
+
+        void UpdateFilter(Func<T, int, IList<T>, bool> newFilter)
+        {
+            if (filter == null && newFilter == null)
+                return; // nothing to do
```

Comment:
If the sorting comparer is reset, it triggers an update to the filter which goes through this method. If the filter is null, it will hit this codepath, and it's valid to just return as a noop.

If the new filter is the same as the existing filter, it will be recalculated as if it's a new filter - again, because if the sorting comparer is changed, everything needs to be recalculated regardless of whether the filter has changed or not.